drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -6142,20 +6142,46 @@ Error RenderingDeviceDriverD3D12::_check_capabilities() {
 	multiview_capabilities.is_supported = false;
 	subgroup_capabilities.size = 0;
 	subgroup_capabilities.wave_ops_supported = false;
-	shader_capabilities.shader_model = D3D_SHADER_MODEL_6_0;
+	shader_capabilities.shader_model = (D3D_SHADER_MODEL)0;
 	shader_capabilities.native_16bit_ops = false;
 	storage_buffer_capabilities.storage_buffer_16_bit_access_is_supported = false;
 	format_capabilities.relaxed_casting_supported = false;
 
-	// Check shader model.
-	D3D12_FEATURE_DATA_SHADER_MODEL shader_model = {};
-	shader_model.HighestShaderModel = MIN(D3D_HIGHEST_SHADER_MODEL, D3D_SHADER_MODEL_6_6);
-	res = device->CheckFeatureSupport(D3D12_FEATURE_SHADER_MODEL, &shader_model, sizeof(shader_model));
-	ERR_FAIL_COND_V_MSG(!SUCCEEDED(res), ERR_CANT_CREATE, "CheckFeatureSupport failed with error " + vformat("0x%08ux", (uint64_t)res) + ".");
+	{
+		static const D3D_SHADER_MODEL SMS_TO_CHECK[] = {
+			D3D_SHADER_MODEL_6_6,
+			D3D_SHADER_MODEL_6_5,
+			D3D_SHADER_MODEL_6_4,
+			D3D_SHADER_MODEL_6_3,
+			D3D_SHADER_MODEL_6_2,
+			D3D_SHADER_MODEL_6_1,
+			D3D_SHADER_MODEL_6_0, // Determined by NIR (dxil_min_shader_model).
+		};
+
+		auto _d3d_sm_to_str = [](D3D_SHADER_MODEL p_sm) {
+			return vformat("%d.%d", (p_sm >> 4), (p_sm & 0xf));
+		};
```

Comment:
I'd use a macro instead of a lambda for this

Reply:
If this isn't within the reasonable usage of lambdas, I don't know what is. I've changed it, but I'm curious about what's your rationale here.

Reply:
It doesn't really benefit from being a method I'd say, it doesn't do anything a macro can't do in this case

Reply:
I guess in the future we'll be able to come up with a ruleset that lets us take advantage of lambdas in a reasonable manner, without the current uncertainty about what the review outcome will be. I believe lambdas can make the codebase better if wisely used. The ability to tell that something is a lambda by its name seems indeed like an interesting part of that.

Reply:
Yeah I was thinking of a naming standard like prefixing with `l_` or `f_`, that would help with that part